### PR TITLE
CLOUDSTACK-9972: Enhance listVolume API to include physical size and …

### PR DESCRIPTION
**Pull request by abhinandanprateek**
…utilization.

Also fixed pool, cluster and pod info

### CODE REVIEW

**Comment by cloudmonger** [comment]
### ACS CI BVT Run
 **Sumarry:**
 Build Number 909
 Hypervisor xenserver
 NetworkType Advanced
 Passed=86
 Failed=18
 Skipped=12

_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/r2si930m8xxzavs/AAAzNrnoF1fC3auFrvsKo_8-a?dl=0

**Failed tests:**
* test_vm_snapshots.py

 * test_change_service_offering_for_vm_with_snapshots Failed

* test_scale_vm.py

 * ContextSuite context=TestScaleVm>:setup Failing since 21 runs

* test_reset_vm_on_reboot.py

 * test_01_reset_vm_on_reboot Failed

* test_non_contigiousvlan.py

 * test_extendPhysicalNetworkVlan Failed

* test_deploy_vm_iso.py

 * test_deploy_vm_from_iso Failing since 2 runs

* test_list_ids_parameter.py

 * ContextSuite context=TestListIdsParams>:setup Failing since 31 runs

* test_metrics_api.py

 * test_list_volumes_metrics Failed

* test_volumes.py

 * test_01_create_volume Failed

 * ContextSuite context=TestVolumes>:setup Failing since 14 runs

* test_ssvm.py

 * test_06_stop_cpvm Failed

 * test_08_reboot_cpvm Failed

* test_deploy_vm_root_resize.py

 * test_00_deploy_vm_root_resize Failed

* test_resource_detail.py

 * ContextSuite context=TestResourceDetail>:setup Failed

* test_vm_life_cycle.py

 * test_10_attachAndDetach_iso Failing since 2 runs

* test_routers_network_ops.py

 * test_01_isolate_network_FW_PF_default_routes_egress_true Failing since 35 runs

 * test_02_isolate_network_FW_PF_default_routes_egress_false Failing since 35 runs

 * test_01_RVR_Network_FW_PF_SSH_default_routes_egress_true Failing since 35 runs

 * test_02_RVR_Network_FW_PF_SSH_default_routes_egress_false Failing since 35 runs


**Skipped tests:**
test_vm_nic_adapter_vmxnet3
test_01_verify_libvirt
test_02_verify_libvirt_after_restart
test_03_verify_libvirt_attach_disk
test_04_verify_guest_lspci
test_05_change_vm_ostype_restart
test_06_verify_guest_lspci_again
test_static_role_account_acls
test_11_ss_nfs_version_on_ssvm
test_nested_virtualization_vmware
test_3d_gpu_support
test_deploy_vgpu_enabled_vm

**Passed test suits:**
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_over_provisioning.py
test_global_settings.py
test_service_offerings.py
test_routers_iptables_default_policy.py
test_loadbalance.py
test_routers.py
test_deploy_vms_with_varied_deploymentplanners.py
test_network.py
test_router_dns.py
test_login.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_nic.py
test_secondary_storage.py
test_disk_offerings.py

**Comment by borisstoyanov** [comment]
@blueorangutan package

**Comment by borisstoyanov** [comment]
@blueorangutan test

**Comment by borisstoyanov** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
@DaanHoogland @nvazquez  /cc @borisstoyanov -- can one of your pick up this PR and fix issues, refactor configs and help towards testing.

**Comment by nvazquez** [comment]
Sure @rhtyd, I can pick it up if you don't mind guys @DaanHoogland @borisstoyanov

**Comment by rhtyd** [comment]
@nvazquez thanks that would be great

**Comment by nvazquez** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan test

**Comment by nvazquez** [comment]
@rhtyd issues fixed, tests failing on Travis seems unrelated: `"CloudstackAPIException: Execute cmd: createproject failed, due to: errorCode: 535, errorText:Maximum number of resources of type 'project' for domain id=5 has been exceeded.\n"]`

**Comment by rhtyd** [comment]
Thanks @nvazquez LGTM, ignoring intemittent/known failures, the tests also LGTM.

**Comment by rhtyd** [comment]
alright, we need to run another round of testing and some new changes /cc @nvazquez

**Comment by nvazquez** [comment]
@rhtyd I added changes

**Comment by nvazquez** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@rhtyd @borisstoyanov I've rebased master due to recent merge on master caused conflicts, I'll kick package again and then test on BO

**Comment by nvazquez** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan test

**Comment by cloudmonger** [comment]
### ACS CI BVT Run
 **Sumarry:**
 Build Number 1170
 Hypervisor xenserver
 NetworkType Advanced
 Passed=117
 Failed=4
 Skipped=40

_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/r2si930m8xxzavs/AAAzNrnoF1fC3auFrvsKo_8-a?dl=0

**Failed tests:**
* test_routers_network_ops.py

 * test_01_isolate_network_FW_PF_default_routes_egress_true Failing since 34 runs

 * test_02_isolate_network_FW_PF_default_routes_egress_false Failing since 161 runs

 * test_01_RVR_Network_FW_PF_SSH_default_routes_egress_true Failing since 156 runs

 * test_02_RVR_Network_FW_PF_SSH_default_routes_egress_false Failing since 156 runs


**Skipped tests:**
test_vm_nic_adapter_vmxnet3
test_01_verify_libvirt
test_02_verify_libvirt_after_restart
test_03_verify_libvirt_attach_disk
test_04_verify_guest_lspci
test_05_change_vm_ostype_restart
test_06_verify_guest_lspci_again
test_disable_oobm_ha_state_ineligible
test_ha_kvm_host_degraded
test_ha_kvm_host_fencing
test_ha_kvm_host_recovering
test_hostha_configure_default_driver
test_hostha_enable_ha_when_host_disabled
test_hostha_enable_ha_when_host_disconected
test_hostha_enable_ha_when_host_in_maintenance
test_remove_ha_provider_not_possible
test_configure_ha_provider_invalid
test_configure_ha_provider_valid
test_ha_configure_enabledisable_across_clusterzones
test_ha_disable_feature_invalid
test_ha_enable_feature_invalid
test_ha_list_providers
test_ha_multiple_mgmt_server_ownership
test_ha_verify_fsm_available
test_ha_verify_fsm_degraded
test_ha_verify_fsm_fenced
test_ha_verify_fsm_recovering
test_hostha_configure_default_driver
test_hostha_configure_invalid_provider
test_hostha_disable_feature_valid
test_hostha_enable_feature_valid
test_hostha_enable_feature_without_setting_provider
test_list_ha_for_host
test_list_ha_for_host_invalid
test_list_ha_for_host_valid
test_static_role_account_acls
test_11_ss_nfs_version_on_ssvm
test_nested_virtualization_vmware
test_3d_gpu_support
test_deploy_vgpu_enabled_vm

**Passed test suits:**
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_vm_snapshots.py
test_over_provisioning.py
test_global_settings.py
test_router_dnsservice.py
test_scale_vm.py
test_service_offerings.py
test_routers_iptables_default_policy.py
test_loadbalance.py
test_routers.py
test_reset_vm_on_reboot.py
test_deploy_vms_with_varied_deploymentplanners.py
test_network.py
test_router_dns.py
test_outofbandmanagement_nestedplugin.py
test_non_contigiousvlan.py
test_login.py
test_deploy_vm_iso.py
test_list_ids_parameter.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_metrics_api.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_volumes.py
test_nic.py
test_deploy_vm_root_resize.py
test_resource_detail.py
test_secondary_storage.py
test_vm_life_cycle.py
test_disk_offerings.py

**Comment by nvazquez** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan test

**Comment by cloudmonger** [comment]
### ACS CI BVT Run
 **Sumarry:**
 Build Number 1180
 Hypervisor xenserver
 NetworkType Advanced
 Passed=114
 Failed=9
 Skipped=40

_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/r2si930m8xxzavs/AAAzNrnoF1fC3auFrvsKo_8-a?dl=0

**Failed tests:**
* test_router_dnsservice.py

 * test_router_dns_guestipquery Failed

* test_scale_vm.py

 * ContextSuite context=TestScaleVm>:setup Failing since 42 runs

* test_portforwardingrules.py

 * test_01_create_delete_portforwarding_fornonvpc Failed

 * test_01_create_delete_portforwarding_fornonvpc Failing since 2 runs

* test_routers_network_ops.py

 * test_01_isolate_network_FW_PF_default_routes_egress_true Failing since 41 runs

 * test_02_isolate_network_FW_PF_default_routes_egress_false Failing since 168 runs

 * test_01_RVR_Network_FW_PF_SSH_default_routes_egress_true Failing since 163 runs

 * test_02_RVR_Network_FW_PF_SSH_default_routes_egress_false Failing since 163 runs

 * test_03_RVR_Network_check_router_state Failed


**Skipped tests:**
test_vm_nic_adapter_vmxnet3
test_01_verify_libvirt
test_02_verify_libvirt_after_restart
test_03_verify_libvirt_attach_disk
test_04_verify_guest_lspci
test_05_change_vm_ostype_restart
test_06_verify_guest_lspci_again
test_disable_oobm_ha_state_ineligible
test_hostha_configure_default_driver
test_hostha_enable_ha_when_host_disabled
test_hostha_enable_ha_when_host_disconected
test_hostha_enable_ha_when_host_in_maintenance
test_hostha_kvm_host_degraded
test_hostha_kvm_host_fencing
test_hostha_kvm_host_recovering
test_remove_ha_provider_not_possible
test_configure_ha_provider_invalid
test_configure_ha_provider_valid
test_ha_configure_enabledisable_across_clusterzones
test_ha_disable_feature_invalid
test_ha_enable_feature_invalid
test_ha_list_providers
test_ha_multiple_mgmt_server_ownership
test_ha_verify_fsm_available
test_ha_verify_fsm_degraded
test_ha_verify_fsm_fenced
test_ha_verify_fsm_recovering
test_hostha_configure_default_driver
test_hostha_configure_invalid_provider
test_hostha_disable_feature_valid
test_hostha_enable_feature_valid
test_hostha_enable_feature_without_setting_provider
test_list_ha_for_host
test_list_ha_for_host_invalid
test_list_ha_for_host_valid
test_static_role_account_acls
test_11_ss_nfs_version_on_ssvm
test_nested_virtualization_vmware
test_3d_gpu_support
test_deploy_vgpu_enabled_vm

**Passed test suits:**
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_vm_snapshots.py
test_over_provisioning.py
test_global_settings.py
test_service_offerings.py
test_routers_iptables_default_policy.py
test_loadbalance.py
test_routers.py
test_reset_vm_on_reboot.py
test_deploy_vms_with_varied_deploymentplanners.py
test_network.py
test_router_dns.py
test_outofbandmanagement_nestedplugin.py
test_non_contigiousvlan.py
test_login.py
test_deploy_vm_iso.py
test_list_ids_parameter.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_metrics_api.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_volumes.py
test_nic.py
test_deploy_vm_root_resize.py
test_resource_detail.py
test_secondary_storage.py
test_vm_life_cycle.py
test_disk_offerings.py

**Comment by nvazquez** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan test

**Comment by nvazquez** [comment]
@borisstoyanov @rhtyd issues fixed for this PR, do think ww can proceed merging it?

File: setup/db/db/schema-4930to41000.sql [hunk 1]
```diff
@@ -278,3 +278,140 @@ CREATE TABLE `cloud`.`external_netscaler_controlcenter` (
 
 ALTER TABLE `cloud`.`sslcerts` ADD COLUMN `name` varchar(255) NULL default NULL COMMENT 'Name of the Certificate';
 ALTER TABLE `cloud`.`network_offerings` ADD COLUMN `service_package_id` varchar(255) NULL default NULL COMMENT 'Netscaler ControlCenter Service Package';
+
```
Comment (rhtyd) [inline setup/db/db/schema-4930to41000.sql]: @nvazquez can you move this to 4.10->4.11 sql path?

**Comment by rhtyd** [comment]
@nvazquez sure, can you fix one outstanding issue, after that we may merge it. Thanks.

**Comment by nvazquez** [comment]
Done, thanks @rhtyd

**Comment by rhtyd** [comment]
LGTM, thanks.

**Pull request merged**